### PR TITLE
refactor: redis config

### PR DESCRIPTION
**Pull request by monarch0111**
## Description

This is extension of https://github.com/chatwoot/chatwoot/pull/1301
I missed to add sentinel configuration to Sidekiq. I have refactored redis configuration for application and sidekiq.

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manually tested with Sidekiq and Rails
- Added RSpec to check return values for Redis::Config

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by monarch0111** [comment]
@sojan-official Apologies for additional PR, I realized that I missed to update Sidekiq Redis Sentinel config in my last PR.

File: lib/redis/config.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if const_defined? 'BASE_CONFIG'
+          BASE_CONFIG
+        else
+          reload_config
+        end
+      end
+      config.merge(size: SIDEKIQ_SIZE)
+    end
+
+    def reload_config
```
Comment (monarch0111) [inline lib/redis/config.rb:23]: @sojan-official I added this `reload_config` method to help me change and test configuration under different ENV variable scenarios. Though, I am not feeling very confident of this implementation. Can you suggest a better way?

**Comment by sojan-official** [comment]
> @sojan-official Apologies for additional PR, I realized that I missed to update Sidekiq Redis Sentinel config in my last PR.

Even I missed it. thanks a ton for catching that :)

File: lib/redis/config.rb (hunk 1 above)
Comment (sojan-official) [reply lib/redis/config.rb:23]: looks good to me at the first glance. 
@sony-mathew  could you take a look ?

Comment (sony-mathew) [reply lib/redis/config.rb:23]: We can definitely refactor this. But if develop is broken right now and this fixes it, should we merge this and then refactor?

Comment (monarch0111) [reply lib/redis/config.rb:23]: @sony-mathew Redis Sentinel feature which was introduced recently is broken, previous redis configuration is okay for app and sidekiq.

Comment (sojan-official) [reply lib/redis/config.rb:23]: @monarch0111  @sony-mathew  I think we can merge this in and do further refactoring in subsequent PRs if required.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**